CMakeLists.txt

```diff
@@ -6,8 +6,19 @@ endif ()
 
 project (dlfcn-win32 C)
 
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")
+include(Macros)
+
 option(BUILD_SHARED_LIBS "shared/static libs" ON) 
 option(BUILD_TESTS "tests?" OFF)
+if(CMAKE_CROSSCOMPILING)
+    add_auto_option(ENABLE_WINE "Enable running tests with wine" AUTO)
+    find_program(WINE_EXECUTABLE wine)
+    check_auto_option(ENABLE_WINE "wine support" WINE_EXECUTABLE "wine executable")
```

Comment:
Logic for usage wine should be: we are not running cmake on windows system (including cygwin) && we are running cmake on unix system && we are cross compiling for windows